cmake/FindLuaJIT.cmake

```diff
@@ -0,0 +1,49 @@
+# Locate a LuaJIT library.
+#
+# This module defines:
+#  LUA_FOUND, if false, do not try to link to Lua,
+#  LUA_LIBRARIES, both lua and lualib,
+#  LUA_INCLUDE_DIR, where to find lua.h,
+#  LUA_VERSION_STRING, the version of Lua found,
+#  LUA_VERSION_MAJOR, the major version of Lua,
+#  LUA_VERSION_MINOR, the minor version of Lua,
+#  LUA_VERSION_PATCH, the patch version of Lua.
+
+set(ERROR_MESSAGE "LuaJIT is not found")
+
+find_package(PkgConfig)
+pkg_check_modules(PC_LUAJIT luajit)
+
+find_path(LUA_INCLUDE_DIR luajit.h
+  HINTS ${PC_LUAJIT_INCLUDEDIR} ${PC_LUAJIT_INCLUDE_DIRS})
+if (STATIC_LUAJIT)
+  find_library(LUA_LIBRARIES NAMES libluajit-5.1.a
+    HINTS ${PC_LUAJIT_LIBDIR} ${PC_LUAJIT_LIBRARY_DIRS})
+else()
+  find_library(LUA_LIBRARIES NAMES libluajit-5.1.so
```

Comment:
I would rather check something like `libluajit-5.1.`[`${CMAKE_SHARED_LIBRARY_SUFFIX}`](https://cmake.org/cmake/help/latest/variable/CMAKE_SHARED_LIBRARY_SUFFIX.html) to work correctly with macOS.